esp-bootloader-esp-idf/src/ota_updater.rs

```diff
@@ -0,0 +1,166 @@
+//! # A more convenient way to access Over The Air Updates (OTA) functionality.
+
+use crate::{
+    ota::OtaImageState,
+    partitions::{AppPartitionSubType, Error, FlashRegion, PartitionTable},
+};
+
+/// This can be used as more convenient - yet less flexible, way to do OTA updates.
+///
+/// If you need lower level access see [crate::ota::Ota]
+#[derive(Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct OtaUpdater<'a, F>
+where
+    F: embedded_storage::Storage,
+{
+    flash: &'a mut F,
+    pt: PartitionTable<'a>,
+    ota_count: usize,
+}
+
+impl<'a, F> OtaUpdater<'a, F>
+where
+    F: embedded_storage::Storage,
+{
+    /// Create a new instance of [OtaUpdater].
+    ///
+    /// # Errors
+    /// [Error::Invalid] if no OTA data partition or less than two OTA app partition were found.
+    pub fn new(
+        flash: &'a mut F,
+        buffer: &'a mut [u8; crate::partitions::PARTITION_TABLE_MAX_LEN],
```

Comment:
I guess we can't just put the buffer inside the updater because then it would be self referential?

Reply:
I think not :D